File: frontend/src/services/busService.js
import api from './api';

export async function getAllBuses() {
  const { data } = await api.get('/buses');
  return data;
}

export async function searchBuses(source, destination, date) {
  const { data } = await api.get('/buses/search', {
    params: { source, destination, date }
  });
  return data;
}

export async function createBus(payload) {
  const { data } = await api.post('/buses', payload);
  return data;
}

export async function updateBus(busId, payload) {
  const { data } = await api.put(`/buses/${busId}`, payload);
  return data;
}

export async function deleteBus(busId) {
  const { data } = await api.delete(`/buses/${busId}`);
  return data;
}

File: backend/src/main/java/com/busbooking/service/BusService.java
package com.busbooking.service;

import com.busbooking.dto.BusRequestDto;
import com.busbooking.dto.BusResponseDto;

import java.time.LocalDate;
import java.util.List;

public interface BusService {
    BusResponseDto createBus(BusRequestDto request);
    List<BusResponseDto> getAllBuses();
    List<BusResponseDto> searchBuses(String source, String destination, LocalDate travelDate);
    BusResponseDto updateBus(String id, BusRequestDto request);
    void deleteBus(String id);
}
